ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/resources/jts/orbspecific/XAResourceRecord.java

```diff
@@ -251,9 +251,14 @@ public org.omg.CosTransactions.Vote prepare() throws HeuristicMixed,
 			case XAException.XA_RBOTHER:
 			case XAException.XA_RBPROTO:
 			case XAException.XA_RBTIMEOUT:
+				// This will avoid calling rollback when aborting
+				_rolledBack = true;
+				return Vote.VoteRollback;
+				// We might need to roll back the XA resource for these error codes
 			case XAException.XAER_INVAL:
 			case XAException.XAER_PROTO:
-			case XAException.XAER_NOTA: // resource may have arbitrarily rolled back (shouldn't, but ...)
+			case XAException.XAER_RMERR:
+			case XAException.XAER_RMFAIL:
 				return Vote.VoteRollback;
```

Comment:
Maybe we need a comment something like:
// We do not know the outcome of the resource manager conclusively and so we will try to call rollback again

Reply:
Sorry, assuming that was there before I missed that. I will resolve this.